src/fsharp/NameResolution.fs

```diff
@@ -2905,16 +2914,34 @@ let ResolveCompletionsInType (ncenv: NameResolver) nenv (completionTargets: Reso
                 AllMethInfosOfTypeInScope ncenv.InfoReader nenv (None,ad) PreferOverrides m typ 
                 |> List.filter minfoFilter
 
-            let addersAndRemovers = 
-                pinfoItems 
-                |> List.map (function Item.Event(FSEvent(_,_,addValRef,removeValRef)) -> [addValRef.LogicalName;removeValRef.LogicalName] | _ -> [])
-                |> List.concat
+            let minfos = 
+                let addersAndRemovers = 
+                    pinfoItems 
+                    |> List.map (function Item.Event(FSEvent(_,_,addValRef,removeValRef)) -> [addValRef.LogicalName;removeValRef.LogicalName] | _ -> [])
+                    |> List.concat
+                    |> set
+
+                if addersAndRemovers.IsEmpty then minfos
+                else minfos |> List.filter (fun minfo -> not (addersAndRemovers.Contains minfo.LogicalName))
+
+#if EXTENSIONTYPING
+            // Filter out the ones with mangled names from applying static parameters
+            let minfos = 
```

Comment:
This filters the mangled forms of statically-instantiated provided methods from the auto-complete lists